web-bundle/src/main/java/com/graphhopper/replica/ReplicaVehicleEncodedValuesFactory.java

```diff
@@ -37,11 +39,36 @@ public VehicleEncodedValues createVehicleEncodedValues(final String vehicleName,
         } else if (vehicleName.equals(RouterConstants.SMALL_TRUCK_VEHICLE_NAME) || baseCustomSpeedsVehicleType == CustomSpeedsVehicle.VehicleType.SMALL_TRUCK) {
             return TruckFlagEncoder.createSmallTruck(vehicleName);
         } else if (vehicleName.equals(RouterConstants.BIKE_VEHICLE_NAME) || baseCustomSpeedsVehicleType == CustomSpeedsVehicle.VehicleType.BIKE) {
-            return VehicleEncodedValues.bike(configuration);
+            if (baseCustomSpeedsVehicleType == CustomSpeedsVehicle.VehicleType.BIKE) {
+                return bike(configuration);
+            } else {
+                return VehicleEncodedValues.bike(configuration);
+            }
         } else if (vehicleName.equals(RouterConstants.FOOT_VEHICLE_NAME) || baseCustomSpeedsVehicleType == CustomSpeedsVehicle.VehicleType.FOOT) {
             return VehicleEncodedValues.foot(configuration);
         }
 
         return super.createVehicleEncodedValues(vehicleName, configuration);
     }
+
+    // Exactly the same as VehicleEncodedValues.bike(), but updated to store speeds in both directions of OSM Ways,
+    // so that SEID-based custom speeds work properly.
+    // Note: we keep storeTwoDirections set to false for the priority encoder. For some reason, storing both directions
+    // here causes odd behavior for other modes (eg the foot routing used in transit requests).
```

Comment:
I still can't explain exactly why this was happening (esp considering it seemed to be affecting foot routing used in transit...), but it was resolved instantly when I stopped storing two directions for bike priority 🤷 

Note that we'll probably get a tiny bit of difference in bike route behavior with the change of `speedTwoDirections` to `true`, but my guess is it just makes bike routing more accurate, because speeds are now more granular for both directions of each OSM Way